### ISSUE TITLE
Error with Journal Multi Currency Lines

### ISSUE THREAD

**Issue by e-Evolution** [adempiere/adempiere#1508]
When entry a journal using multi currency the total and currency rate are calculated incorrect.

Case:

1.- Entry Batch Journal and Journal using USD currency
2.- Add 2 Journal lines using USD currency 
3.- Add 2 Journal lines using EUR currency 

The total debit and total credit not is rightly  , the current rate for EUR is set as 1 , and is incorrect because the callout define rightly current rate when is entry a line

**Comment by adaxa-steven** [comment]
it seems to me that it is just asking for trouble to allow journal lines to be in different currency. whey do we allow it?

**Comment by e-Evolution** [comment]
@adaxa-steven is very common when you add to beginning balance to starting a implementation that somes balance for accounts are in different currency , in this case for instance bank account.

**Comment by adaxa-steven** [comment]
The bank may be in foreign currency but GL account will be in the accounted currency.  you could still enter a 2 line journal in the foreign currency dB bank credit suspense account into bank cuuency and then the whole TB in accounting currency with an offsetting debit to the suspense account.
This would ensure that the journal debits equalled thejournal credits exactly..  which is never a bad thing ;-)

**Comment by adaxa-steven** [comment]
The bank may be in foreign currency but GL account will be in the accounted currency.  you could still enter a 2 line journal in the foreign currency dB bank credit suspense account into bank cuuency and then the whole TB in accounting currency with an offsetting debit to the suspense account.
This would ensure that the journal debits equalled thejournal credits exactly..  which is never a bad thing ;-)

**Comment by e-Evolution** [comment]
@adaxa-steven yes , use suspense account may be a technical solution , but my accountant not understand the reason of not load your balances with the last close period in this case for december 2017.

On the other hand I need entry the source amount in USD when my accounting schema use MNX because need know always the balance in USD for each account in foreign currency, because will use for the process to calculate the not realized  Gain/Lost 

if now you can set some lines in other currency and use rate to get the account debit or credit.

**Comment by adaxa-steven** [comment]
Its not an issue for me .. I just make currency read only on the line and inherit the currency from the journal.
It becomes an issue for others without that change when someone enters debit line 1 in currency 1 and credit line 2 in currency 2 and then the debits <> credits... and then they want to be told how to fix the mess.

**Comment by e-Evolution** [comment]
@adaxa-steven do you are right all document only have a currency by document, in mi case I can solve entry other Journal using the other currency.

**Comment by yamelsenih** [comment]
Hi @e-Evolution have you a pull request for it?
